index.js

```diff
@@ -22,35 +16,31 @@ app.use(express.json());
 app.use(express.static("public"));
 
 io.on("connection", (socket) => {
-  const handleSignin = (args) => {
-    const { lat, lon } = args || {};
+  let { hash: userHash } = socket.handshake.query;
 
-    if (!lat || !lon) {
-      return;
-    }
-
-    const hash = geohash.encode(lat, lon, 5);
-
-    const randomName = uniqueNamesGenerator({
-      dictionaries: [adjectives, colors, animals],
-      separator: "-",
-    }); // big-red-donkey
-
-    let grid = grids[hash];
+  if (userHash) {
+    // Join the user to the grid
+    socket.join(userHash);
 
-    if (!grid) {
-      grids[hash] = new Set();
-      grid = grids[hash];
+    // Send all users in the grid to the newly connected user
+    if (grids[userHash]) {
+      io.to(userHash).emit("users", [...grids[userHash]]);
     }
+  }
 
-    grid.add({
-      randomName,
-    });
+  socket.on(
+    "signin",
+    handleSignin(socket, grids, (result) => {
+      const { hash } = result || {};
+      userHash = hash;
 
-    socket.emit("signin", { hash, username: randomName });
-  };
+      // Join the user to the grid
+      socket.join(hash);
 
-  socket.on("signin", handleSignin);
+      // Send all users in the grid to the newly connected user
+      io.to(hash).emit("users", [...grids[userHash]]);
```

Comment:
What is the significance of accessing the same value with two different names in this function ?

Reply:
There's no benefit. I'll update the names real quick to eliminate any confusion.